### ISSUE TITLE
ListModel could support iterables that have no __getitem__

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7169]
|<img alt="Tv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Tv reported|
|-|-|
|Trac ID|trac#355|
|Type|enhancement|
|Created|2003-10-19 18:53:20Z|

<details><summary>Searchable metadata</summary>

```
trac-id__355 355
type__enhancement enhancement
reporter__Tv Tv
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__web web
time__1066589600000000 1066589600000000
changetime__1468914449784029 1468914449784029
version__ 
owner__ 
cc__Tv
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Tv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Tv commented|
|-|-|

```
#!html
<pre>
With the following change, ListModel could be used with e.g. Sets and other
iterables. It's not pretty, I know -- think of it more as a use case than as a
complete solution. I want to iterate over the elements in a Set.

Index: twisted/web/woven/model.py
===================================================================
RCS file: /cvs/Twisted/twisted/web/woven/model.py,v
retrieving revision 1.51
diff -u -u -r1.51 model.py
--- twisted/web/woven/model.py	6 Oct 2003 15:10:17 -0000	1.51
+++ twisted/web/woven/model.py	19 Oct 2003 12:39:07 -0000
@@ -329,7 +329,11 @@
             return None
         if i > len(orig):
             return None
-        sm = adaptToIModel(orig[i], self, name)
+        try:
+            item = orig[i]
+        except AttributeError, e:
+            item = list(orig)[i]
+        sm = adaptToIModel(item, self, name)
         self.submodels[name] = sm
         return sm
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

```
#!html
<pre>
This bug is being resolved, wontfix, as woven is "almost deprecated" and unmaintained. Use nevow 
instead. See &lt;http://divmod.org/users/dp.twistd/wiki/> and the twisted-web mailing list &lt;http://
twistedmatrix.com/cgi-bin/mailman/listinfo/twisted-web>.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub commented|
|-|-|

In changeset bd18732299cbf4438d774a4a866b6b286891304f
```
#!CommitTicketReference repository="" revision="bd18732299cbf4438d774a4a866b6b286891304f"
Merge pull request #355 from twisted/8585-scop-ircconstants

Author: scop
Reviewer: adiroiban, rodrigc
Fixes: #8585

Add IRC constants specified in RFC 2812 and errata.
```
